petclinic-frontend/src/features/customers/components/AddPetForm.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import { FormEvent, useState } from 'react';
+import { addPet } from '../api/addPet';
+import { PetRequestModel } from '../models/PetRequestModel';
+import { useNavigate, useParams } from 'react-router-dom';
+import './AddPetForm.css';
+
+const petTypeOptions: { [key: string]: string } = {
+  '1': 'Cat',
+  '2': 'Dog',
+  '3': 'Lizard',
+  '4': 'Snake',
+  '5': 'Bird',
+  '6': 'Hamster',
+};
+
+const AddPetForm: React.FC = (): JSX.Element => {
+  const navigate = useNavigate();
+  const { ownerId } = useParams<{ ownerId: string }>();
+  const [pet, setPet] = useState<PetRequestModel>({
+    ownerId: ownerId || '',
```

Comment:
Why is it ownerId or nothing ? Could it be empty? Is that the correct way to handle it?

Reply:
It was to fix an error, yea we can try to change the logic, I agree it can be wonky